context.go

```diff
@@ -778,7 +794,22 @@ loop:
 		case dep := <-depsCh:
 			deps = append(deps, dep)
 		case file := <-fileCh:
-			visitor(file)
+			syncVisitor(file)
+			go func() {
+				// wait for ancestor
+				<-visitCompletionChs[ancestorsMap[file.Name]]
```

Comment:
Instead of looking up in two maps you could wrap the parser.File in a new type that holds a pointer to the ancestor (computed from descendantsMap) and a channel.  Then you wouldn't need ancestorsMap at all.

Reply:
ok